Docs for primitive key support (#4478)
confluentinc/ksql: docs-md/developer-guide/syntax-reference.md

Example:

```sql

> CREATE TABLE users (rowkey INT KEY, registertime BIGINT, gender VARCHAR, regionid VARCHAR, userid INT)

Reviewer: all the examples have added `ROWKEY <TYPE> KEY` - is this now a requirement or is it just illustrative?

Author: It's not currently required. If you don't add one KSQL currently defaults to `ROWKEY STRING KEY`.  However, we should encourage people to be explicit, rather than rely on implicit.

I've not documented this yet as its a bit meh.  Ideally, if you don't supply the key column it should mean there is no key column.  We can update the docs when this is the case.

-   Goal: You want to create a table from a topic, which is keyed by
    userid of type INT.
-   Problem: The required key is present as a field/column (aptly named

>     userid) in the message value but the actual message key in Kafka is

Reviewer 2: ```suggestion
    `userid`) in the message value, but the actual message key in {{ site.ak }} is
```

    not set or has some other value or format.

```sql
-- Create a stream on the original topic
CREATE STREAM users_with_wrong_key (userid INT, username VARCHAR, email VARCHAR)
  WITH (KAFKA_TOPIC='users', VALUE_FORMAT='JSON');
  
-- Derive a new stream with the required key changes.
-- 1) The CAST statement converts the key to the required format.
-- 2) The PARTITION BY clause re-partitions the stream based on the new, converted key.
-- The resulting schema will be: ROWKEY INT, USERNAME STRING, EMAIL STRING
-- the userId will be stored in ROWKEY.
CREATE STREAM users_with_proper_key
  WITH(KAFKA_TOPIC='users-with-proper-key') AS
  SELECT username, email
  FROM users_with_wrong_key
  PARTITION BY userid
  EMIT CHANGES;

-- Now you can create the table on the properly keyed stream.
CREATE TABLE users_table (ROWKEY INT KEY, username VARCHAR, email VARCHAR)
  WITH (KAFKA_TOPIC='users-with-proper-key',
        VALUE_FORMAT='JSON');

> -- Or, if you would prefer, you can keep userId in the value of the repartitioned data

Reviewer 2: ```suggestion
-- Or, if you prefer, you can keep userId in the value of the repartitioned data
```

> -- This would allow you to use the more descriptive `userId` rather than ROWTIME.

Reviewer 2: ```suggestion
-- This enables using the more descriptive `userId` rather than ROWTIME.
```
